blackjax/hmc.py

```diff
@@ -0,0 +1,117 @@
+"""Public API for the HMC Kernel"""
+from typing import Callable, NamedTuple, Union
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+import blackjax.inference.base as base
+import blackjax.inference.integrators as integrators
+import blackjax.inference.metrics as metrics
+import blackjax.inference.proposals as proposals
+
+Array = Union[np.array, jnp.DeviceArray]
+
+
+class HMCParameters(NamedTuple):
+    step_size: float
+    inv_mass_matrix: Array
+    num_integration_steps: int
+
+
+def new_states(position: jax.pytree, logpdf: Callable):
+    """Create chain states from positions.
+
+    The HMC kernel works with states that contain the current chain position
+    and its associated potential energy and derivative of the potential energy.
+    This function computes initial states from an initial position.
+
+    While this function is intended to work for one chain, it is possible to
+    use `jax.vmap` to compute the initial state of several chains.
+
+    Example
+    -------
+    Let us assume a model with two random variables. We wish to sample from
+    4 chains with the following initial positions:
+
+        >>> import numpy as np
+        >>> init_positions = (np.random.rand(4, 1000), np.random.rand(4, 300))
+
+    We have a `logpdf` function that returns the log-probability associated with
+    the chain at a given position:
+
+        >>> logpdf((np.random.rand(1000), np.random.rand(3000)))
+        -3.4
+
+    We can compute the initial state for each of the 4 chain as follows:
+
+        >>> import jax
+        >>> jax.vmap(new_states, in_axes=(0, None))(init_positions, logpdf)
+
+    Parameters
+    ----------
+    position
+        The current values of the random variables whose posterior we want to
+        sample from. Can be anything from a list, a (named) tuple or a dict of
+        arrays. The arrays can either be Numpy arrays or JAX DeviceArrays.
```

Comment:
Granted I'm new to `jax`, so please ignore this comment if it makes no sense, but are these types (list, named tuple, etc.) consistent with the `jax.pytree` annotation?

Reply:
These are the pytrees with which primitives such as `value_and_grad` will work.